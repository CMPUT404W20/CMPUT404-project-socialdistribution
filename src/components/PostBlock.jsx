import React, { Component } from "react";
import "../styles/Post.scss";
import moreIcon from "../staticfiles/more-icon.svg";
import demoImage from "../staticfiles/demo-img.png";

class PostBlock extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      <div className="post-block">
        <div className="post-info">
          <div className="post-username">
            test user
          </div>
          <div>
            <img id="post-more-icon" src={moreIcon} alt="more-icon"/>
          </div>
          <div className="post-time">
            10 hours ago
          </div>
        </div>
        <img className="post-img" src={demoImage} alt="more-icon"/> 
        <div className="post-content">
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
        </div>
      </div>
    );
  }
}

export default PostBlock;
