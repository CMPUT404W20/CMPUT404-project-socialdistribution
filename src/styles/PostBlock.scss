$BACKGROUND-WHITE: #ffffff;
$left: 25px;
$right: 25px;
$top: 15px;
$bottom: 15px;
$post-width: 500px;
$post-info-font-size: 12px;

.post-block {
  width: $post-width;
  height: fit-content;
  background-color: $BACKGROUND-WHITE;
  box-shadow: 0px 0px 6px rgba(202, 202, 202, 0.25);
}
.post-username,
.post-time {
  display: inline;
  padding-top: 2.5px;
}

.post-info {
  padding: $top $right $bottom $left;
  margin-bottom: $bottom + 10px;
  font-size: $post-info-font-size;
}

.post-username {
  text-align: left;
  float: left;
}
.post-time {
  text-align: right;
  color: #aaaaaa;
  margin-right: 0vw;
  float: right;
}
.post-img {
  width: 100%;
}
.post-content {
  text-align: left;
  padding: $top $right $bottom $left;
  line-height: normal;
}

#post-button {
  all: unset;
  border: none;
  outline: none;
  float: right;
  margin-left: 10px;
}

.dropdown-toggle::after {
  border: none;
}
