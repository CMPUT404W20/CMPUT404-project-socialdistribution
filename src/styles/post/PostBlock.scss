$background-color: #ffffff;
$left: 25px;
$right: 25px;
$top: 15px;
$bottom: 15px;
$post-width: 500px;
$post-info-font-size: 12px;

.post-block {
  width: $post-width;
  height: fit-content;
  background-color: $background-color;
  box-shadow: 0px 0px 6px rgba(202, 202, 202, 0.25);
}

.post-info {
  padding: 0 25px;
  height: 30pt;
  line-height: 30pt;
  font-size: $post-info-font-size;
}

.post-user-and-visibility {
  text-align: left;
  float: left;
  vertical-align: middle;
  position: relative;
  top: 0.5pt;

  svg {
    position: relative;
    top: 1.5pt;
    left: 2pt;
    height: 100%;
  }
}
.post-time {
  color: #aaaaaa;
  float: right;
}
.post-img {
  width: 100%;
}
.post-content {
  text-align: left;
  padding: $top $right $bottom $left;
  line-height: normal;
}

#post-more-button {
  all: unset;
  border: none;
  outline: none;
  float: right;
  margin-left: 10px;
}

.dropdown-menu[aria-labelledby="post-button"] {
  border: 1px solid #5e5e5e;
  border-radius: 2px;
}

.dropdown-toggle::after {
  border: none;
}
