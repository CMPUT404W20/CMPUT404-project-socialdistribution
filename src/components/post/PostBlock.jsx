import React, { Component } from "react";
import PropTypes from "prop-types";
import "../../styles/post/PostBlock.scss";
import DropdownButton from "react-bootstrap/DropdownButton";
import PostDropDown from "./PostDropDown";
import moreIcon from "../../images/more-icon.svg";

class PostBlock extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {
      username, content, imageSrc, postTime,
    } = this.props;
    return (
      <div className="post-block">
        <div className="post-info">
          <div className="post-username">{username}</div>
          <DropdownButton
            id="post-more-button"
            title={<img id="post-more-icon" src={moreIcon} alt="more-icon" />}
            drop="down"
            alignRight
          >
            <PostDropDown />
          </DropdownButton>

          <div className="post-time">{postTime}</div>
        </div>
        <img className="post-img" src={imageSrc} alt="more-icon" />
        <div className="post-content">{content}</div>
      </div>
    );
  }
}

PostBlock.propTypes = {
  username: PropTypes.string.isRequired,
  postTime: PropTypes.string.isRequired,
  imageSrc: PropTypes.node,
  content: PropTypes.string,
};

PostBlock.defaultProps = {
  content: "",
  imageSrc: "",
};

export default PostBlock;
