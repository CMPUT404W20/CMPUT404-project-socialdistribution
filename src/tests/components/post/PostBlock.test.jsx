import React from "react";
import { shallow } from "enzyme";
import PostBlock from "../../../components/post/PostBlock";
import demoImage from "../../../images/demo-img.png";

describe("Post Block Components", () => {
  it("should render correctly", () => {
    const component = shallow(
      <PostBlock
        imageSrc={demoImage}
        username="testuser"
        postTime="10 hours ago"
        content="test blog content"
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
