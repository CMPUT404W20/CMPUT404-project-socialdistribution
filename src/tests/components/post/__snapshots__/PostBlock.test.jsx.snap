// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Block Components should render correctly 1`] = `
<div
  className="post-block"
>
  <div
    className="post-info"
  >
    <span
      className="post-username"
    >
      testuser
    </span>
    <DropdownButton
      alignRight={true}
      drop="down"
      id="post-more-button"
      title={
        <img
          alt="more-icon"
          id="post-more-icon"
          src="more-icon.svg"
        />
      }
    >
      <PostDropDown />
    </DropdownButton>
    <span
      className="post-time"
    >
      a few seconds ago
    </div>
  </div>
  <img
    alt="more-icon"
    className="post-img"
    src="demo-img.png"
  />
  <div
    className="post-content"
  >
    test blog content
  </div>
</div>
`;
